File: quark/test/emit/factorial/py/factorial.py
from quark_runtime import *


def factorial(n):
    if ((n) == (0)):
        return 1;
    else:
        return (n) * (factorial((n) - (1)));



def main():
    _println(factorial(1));
    _println(factorial(2));
    _println(factorial(3));
    _println(factorial(4));
    _println(factorial(5));
    _println(factorial(6));
    _println(factorial(7));
    _println(factorial(8));
    _println(factorial(9));
    _println(factorial(10));


if __name__ == "__main__":
    main()

File: quark/test/emit/factorial/js/factorial.js
var _Q_util = require("util");
function _Q_toString(value) {
    if (value === null) {
        return "null";
    }
    if (Array.isArray(value)) {
        return "[" + value.map(_Q_toString).join(", ") + "]";
    }
    return value.toString();
}

//

function factorial(n) {
    if ((n) === (0)) {
        return 1;
    } else {
        return (n) * (factorial((n) - (1)));
    }
}
exports.factorial = factorial;

function main() {
    console.log(_Q_toString(factorial(1)));
    console.log(_Q_toString(factorial(2)));
    console.log(_Q_toString(factorial(3)));
    console.log(_Q_toString(factorial(4)));
    console.log(_Q_toString(factorial(5)));
    console.log(_Q_toString(factorial(6)));
    console.log(_Q_toString(factorial(7)));
    console.log(_Q_toString(factorial(8)));
    console.log(_Q_toString(factorial(9)));
    console.log(_Q_toString(factorial(10)));
}
exports.main = main;

main();
